### PR TITLE
[SPARK-18828][SPARKR] Refactor scripts for R

### PR DESCRIPTION
**Pull request by felixcheung**
## What changes were proposed in this pull request?

Refactored script to remove duplications and clearer purpose for each script

## How was this patch tested?

manually

### CODE REVIEW

**Comment by felixcheung** [comment]
flaky test?

**Comment by felixcheung** [comment]
Jenkins, retest this please

**Comment by shivaram** [comment]
Can we open a JIRA for this ? Its good to track this change

**Comment by shivaram** [comment]
Just FYI, I'm holding off on this till 2.1 - I think its better to make this change after the release just to be safe

**Comment by felixcheung** [comment]
jenkins, retest this please

**Comment by felixcheung** [comment]
that's weird. but the last Jenkins run on commit 66fc83c passed.
how about we merge this @shivaram ?

**Comment by shivaram** [comment]
Sure I'll take a final look today. Meanwhile

Jenkins, retest this please

**Comment by felixcheung** [comment]
Jenkins, retest this please

**Comment by felixcheung** [comment]
Jenkins, retest this please

**Comment by felixcheung** [comment]
@shivaram how about it? ;)

**Comment by shivaram** [comment]
Sorry for the delay - I was just trying this today. Looks like the `$FWDIR` is not right in `install-dev.sh` ?
```
# cd spark
# ./R/install-dev.sh
Using R_SCRIPT_PATH = /Library/Frameworks/R.framework/Versions/3.2/Resources/bin
/Users/shivaram/spark/R/create-rd.sh: line 34: /Users/shivaram/spark/R/R/find-r.sh: No such file or directory
```

**Comment by felixcheung** [comment]
Hmm, I think it's the problem with running the first script from a relative directory.

./R/install-dev.sh

Vs

cd R
./install-dev.sh

Let me investigate.

**Comment by felixcheung** [comment]
it turns out `$0` is not reliable (http://stackoverflow.com/questions/59895/getting-the-source-directory-of-a-bash-script-from-within). Updated and tested on Ubuntu and Mac.

File: R/install-source-package.sh [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+if [ -z "$VERSION" ]; then
+  VERSION=`grep Version $FWDIR/pkg/DESCRIPTION | awk '{print $NF}'`
+fi
+
+echo "Removing lib path and installing from source package"
+LIB_DIR="$FWDIR/lib"
+rm -rf $LIB_DIR
+mkdir -p $LIB_DIR
+"$R_SCRIPT_PATH/"R CMD INSTALL SparkR_"$VERSION".tar.gz --library=$LIB_DIR
```
Comment (shivaram) [inline R/install-source-package.sh]: Just so I understand correctly this script assumes that `SparkR_$VERSION.tar.gz` exists ? If so can we check that in the preceding lines ?

**Review by shivaram — COMMENTED** [review]
Thanks for the fixes ! local checks work fine now - I just had one last comment

File: R/install-source-package.sh (hunk 1 above)
Comment (felixcheung) [inline R/install-source-package.sh]: great point - added.

File: dev/make-distribution.sh [hunk 2]
```diff
@@ -232,14 +232,17 @@ if [ "$MAKE_R" == "true" ]; then
...
+  NO_TESTS=1 "$SPARK_HOME/"R/check-cran.sh
+
   # Move R source package to match the Spark release version if the versions are not the same.
   # NOTE(shivaram): `mv` throws an error on Linux if source and destination are same file
   if [ "$R_PACKAGE_VERSION" != "$VERSION" ]; then
     mv $SPARK_HOME/R/SparkR_"$R_PACKAGE_VERSION".tar.gz $SPARK_HOME/R/SparkR_"$VERSION".tar.gz
   fi
+
+  # Install source package to get it to generate vignettes rds files, etc.
+  VERSION=$VERSION "$SPARK_HOME/"R/install-source-package.sh
```
Comment (shivaram) [inline dev/make-distribution.sh]: One more thing I ran into while testing this locally - the file `install-source-package.sh` didn't have execute permission. I wonder if this due to the way I patched things or if we should set the executable bit while checking the file in ?

Comment (shivaram) [inline dev/make-distribution.sh]: Actually ignore this - If I use `git apply` it gets the executable bit. I was using `patch` before this.

**Comment by shivaram** [comment]
LGTM

File: dev/make-distribution.sh (hunk 2 above)
Comment (felixcheung) [inline dev/make-distribution.sh]: ah good.

**Comment by felixcheung** [comment]
merging this to master.
AppVeyor seems to be stuck or running behind - since I have PR pending I'll check it in a couple of days.

**Comment by felixcheung** [comment]
merged to master. Will this run in nightly for make-distribution.sh? if not, could you kick off Jenkins as some point? Thanks @shivaram

**Pull request closed without merging**